blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/LocatorTest.java

```diff
@@ -0,0 +1,435 @@
+        locator.isValidDBKey("a.b.c", null);
+
+        // then
+        // the exception is thrown
+    }
+
+    @Test
+    public void isValidDBKeyUndelimitedStringReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertFalse(locator.isValidDBKey("abc", "."));
+    }
+
+    @Test
+    public void isValidDBKeyDelimiterIsNormalCharInDbKeyReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertTrue(locator.isValidDBKey("abc", "b"));
+    }
+
+    @Test
+    public void isValidDBKeyDbKeyContainsDelimiterReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertTrue(locator.isValidDBKey("a.b.c", "."));
+    }
+
+    @Test
+    public void isValidDBKeyEmptyDbKeyReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertFalse(locator.isValidDBKey("", "."));
+    }
+
+    @Test
+    public void isValidDBKeyEmptyDelimiterReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertTrue(locator.isValidDBKey("a.b.c", ""));
+    }
+
+    @Test
+    public void isValidDBKeyMultiCharDelimiterReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertTrue(locator.isValidDBKey("a.b.c", ".b."));
+    }
+
+    @Test
+    public void hashCodeNullReturnsZero() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertEquals(0, locator.hashCode());
+    }
+
+    @Test
+    public void hashCodeReturnsHashCodeOfStringRepresentation() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey(fullyQualifiedName);
+
+        // expect
+        assertEquals(fullyQualifiedName.hashCode(), locator.hashCode());
+    }
+
+    @Test
+    public void equalsNullReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertFalse(locator.equals((Object)null));
+    }
+
+    @Test
+    public void equalsNonLocatorReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertFalse(locator.equals(new Object()));
+    }
+
+    @Test
+    public void equalsDifferentStringReturnsFalse() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey("a.b.c");
+        Locator other = Locator.createLocatorFromDbKey("a.b.d");
+
+        // expect
+        assertFalse(locator.equals((Object)other));
```

Comment:
Yes. There are two `equals` methods, one that takes an `Object` and one that takes a `Locator`. I wanted this test to call the first one, so I explicitly cast to `Object`. Otherwise, it would resolve to the second overload.